zebra/connected.c

```diff
@@ -448,7 +448,7 @@ void connected_down(struct interface *ifp, struct connected *ifc)
 	 * one.
 	 */
 	for (ALL_LIST_ELEMENTS_RO(ifp->connected, cnode, c)) {
-		struct prefix cp;
+		struct prefix cp = {0};
```

Comment:
hmm - it's a bit subtle to have to do this sort of init everywhere someone intends to compare a prefix? is there something we could do to PREFIX_COPY to make it more robust (at least in coverity's opinion)?

Reply:
Good question. I'll look into it.

Reply:
I don't have the coverity details on this, but if it is complaining about a specific function you could signal it as false-positive.

Here is one example: #6251

Reply:
I agree with Mark. From a semantic perspective, I'd expect `PREFIX_COPY` to fully initialize the destination prefix (without the need for pre-initialization).

That said, isn't `PREFIX_COPY` redundant since we already have `prefix_copy()` for the same purpose?

Reply:
@rwestphal right, I had the same thought. There are only a few places in the code where we use PREFIX_COPY(_IPv4/IPv6). They all can be replaced with prefix_copy. I just didn't find time to update the PR yet. I'll try to update this week.